reference_data/management/commands/update_omim.py

```diff
@@ -64,18 +64,41 @@
     4: 'a contiguous gene deletion or duplication syndrome, multiple genes are deleted or duplicated causing the phenotype.',
 }
 
+CACHED_RECORDS_BUCKET = 'seqr-reference-data/omim/'
+CACHED_RECORDS_FILENAME = 'parsed_omim_records.txt'
+CACHED_RECORDS_HEADER = [
+    'gene_id', 'mim_number', 'gene_description', 'comments', 'phenotype_description',
+    'phenotype_mim_number', 'phenotype_map_method', 'phenotype_inheritance', 'phenotypic_series_number',
+]
+
+class CachedOmimReferenceDataHandler(ReferenceDataHandler):
+
+    model_cls = Omim
+    url = 'https://storage.googleapis.com/{bucket}{filename}'.format(
+        filename=CACHED_RECORDS_FILENAME, bucket=CACHED_RECORDS_BUCKET)
+
+    @staticmethod
+    def get_file_header(f):
+        return CACHED_RECORDS_HEADER
+
+    @staticmethod
+    def parse_record(record):
+        yield {k: v or None for k, v in record.items()}
+
 
 class OmimReferenceDataHandler(ReferenceDataHandler):
 
     model_cls = Omim
     url = "https://data.omim.org/downloads/{omim_key}/genemap2.txt"
 
-    def __init__(self, omim_key=None, **kwargs):
+    def __init__(self, omim_key=None, cache_parsed_records=False, **kwargs):
+        """Init OMIM handler."""
         if not omim_key:
             raise CommandError("omim_key is required")
```

Comment:
Since omin_key is required, a `None` default value is not necessary.

Reply:
the signature for the class inheritance requires this to be a keyword arg and not a regular arg, and thus a default is required